### PR TITLE
examples: use grpc v1.30.0 rather than v1.30.0-dev.1

### PR DESCRIPTION
**Pull request by chalin**
@dfawley

### CODE REVIEW

File: examples/go.mod [hunk 1]
```diff
@@ -3,9 +3,7 @@ module google.golang.org/grpc/examples
...
 
 require (
-	github.com/golang/mock v1.1.1
-	github.com/golang/protobuf v1.4.0
-	golang.org/x/oauth2 v0.0.0-20180821212333-d2e6202438be
-	google.golang.org/genproto v0.0.0-20190819201941-24fa4b261c55
-	google.golang.org/grpc v1.30.0-dev.1
+	github.com/golang/protobuf v1.4.1
+	google.golang.org/grpc v1.30.0
+	google.golang.org/protobuf v1.25.0
```
Comment (menghanl) [inline examples/go.mod:8]: Did you add this manually? Or it was go mod tidy?

We now have both old and new protobuf.
I don't think the examples use the new protobuf.

Comment (chalin) [reply examples/go.mod:8]: Manually cleaned out the `go.mod` file leaving only the `grpc` module, and then ran the tests using `examples_test.sh`. What you see here is the `go.mod` file that results.

Running `go mod tidy` on this PR results in no changes.

Comment (menghanl) [reply examples/go.mod:8]: Where did you run `go mod tidy`? Note that you need to `cd examples` and then run it. It gives me
```go
module google.golang.org/grpc/examples

go 1.11

require (
	github.com/golang/mock v1.4.3
	github.com/golang/protobuf v1.4.1
	golang.org/x/oauth2 v0.0.0-20180821212333-d2e6202438be
	google.golang.org/genproto v0.0.0-20200526211855-cb27e3aa2013
	google.golang.org/grpc v1.30.0
	google.golang.org/protobuf v1.25.0 // indirect
)
```

---

And `google.golang.org/protobuf` is a dependency
```
$ go mod why -m google.golang.org/protobuf
# google.golang.org/protobuf
google.golang.org/grpc/examples/features/errors/client
google.golang.org/genproto/googleapis/rpc/errdetails
google.golang.org/protobuf/reflect/protoreflect

```

Comment (menghanl) [reply examples/go.mod:8]: What's even more weird is, running `go get -u google.golang.org/grpc` from HEAD gives
```go
module google.golang.org/grpc/examples

go 1.11

require (
	github.com/golang/mock v1.1.1
	github.com/golang/protobuf v1.4.2
	golang.org/x/net v0.0.0-20200625001655-4c5254603344 // indirect
	golang.org/x/oauth2 v0.0.0-20180821212333-d2e6202438be
	golang.org/x/sys v0.0.0-20200622214017-ed371f2e16b4 // indirect
	golang.org/x/text v0.3.3 // indirect
	google.golang.org/genproto v0.0.0-20200624020401-64a14ca9d1ad
	google.golang.org/grpc v1.30.0
	google.golang.org/protobuf v1.25.0 // indirect
)
```

Note that it doesn't update `mock`, but adds a bunch of other depdencies.

Comment (dfawley) [reply examples/go.mod:8]: As long as you have #3709 in your branch, then we should be able to trust all `go.mod` contents if travis is green.

**Comment by chalin** [comment]
Rebased and started afresh; PTAL.

**Comment by chalin** [comment]
Ready to be merged?

**Pull request merged**